### ISSUE TITLE
IllegalargumentException in NextcloudArrayDeserializer

### ISSUE THREAD

**Issue by hjuergens** [stefan-niedermann/nextcloud-deck#824]
Right after i select my account i  get the following message:

```
App Version: 1.14.1
App Version Code: 1014001
Server App Version: 1.2.4
App Flavor: fdroid

Files App Version Code: 30150090

---

OS Version: 3.18.14-13997403-QB34852243(A320FLXXS9CTJ2)
OS API Level: 26
Device: a3y17lte
Manufacturer: samsung
Model (and Product): SM-A320FL (a3y17ltexc)

---

java.lang.IllegalArgumentException: NextcloudArrayDeserializer got an non-array Json Object: {"1":{"title":"Personal","owner":{"primaryKey":"juergens","uid":"juergens","displayname":"juergens","type":0},"color":"0087C5","archived":false,"labels":[{"title":"Finished","color":"31CC7C","boardId":1,"cardId":null,"lastModified":1612631775,"id":1,"ETag":"3728c65f07362e80085fb90c90e342b0"},{"title":"To review","color":"317CCC","boardId":1,"cardId":null,"lastModified":1612631775,"id":2,"ETag":"3728c65f07362e80085fb90c90e342b0"},{"title":"Action needed","color":"FF7A66","boardId":1,"cardId":null,"lastModified":1612631775,"id":3,"ETag":"3728c65f07362e80085fb90c90e342b0"},{"title":"Later","color":"F1DB50","boardId":1,"cardId":null,"lastModified":1612631775,"id":4,"ETag":"3728c65f07362e80085fb90c90e342b0"}],"acl":[],"permissions":{"PERMISSION_READ":true,"PERMISSION_EDIT":true,"PERMISSION_MANAGE":true,"PERMISSION_SHARE":true},"users":[{"primaryKey":"juergens","uid":"juergens","displayname":"juergens","type":0}],"stacks":[],"deletedAt":0,"lastModified":1612631775,"settings":{"notify-due":"assigned","calendar":true},"id":1,"ETag":"3728c65f07362e80085fb90c90e342b0"}}
	at it.niedermann.nextcloud.deck.api.NextcloudArrayDeserializer.deserialize(NextcloudArrayDeserializer.java:39)
	at it.niedermann.nextcloud.deck.api.NextcloudArrayDeserializer.deserialize(NextcloudArrayDeserializer.java:18)
	at com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:69)
	at com.google.gson.Gson.fromJson(Gson.java:932)
	at com.google.gson.Gson.fromJson(Gson.java:897)
	at com.nextcloud.android.sso.api.NextcloudAPI.convertStreamToTargetEntity(NextcloudAPI.java:135)
	at com.nextcloud.android.sso.api.NextcloudAPI.lambda$performRequestObservableV2$1$NextcloudAPI(NextcloudAPI.java:111)
	at com.nextcloud.android.sso.api.-$$Lambda$NextcloudAPI$8WS9mapJbOYsPgbhBi25vBNYZqI.subscribe(Unknown Source:6)
	at io.reactivex.internal.operators.observable.ObservableFromPublisher.subscribeActual(ObservableFromPublisher.java:31)
	at io.reactivex.Observable.subscribe(Observable.java:12284)
	at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeTask.run(ObservableSubscribeOn.java:96)
	at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
	at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
	at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
	at java.lang.Thread.run(Thread.java:764)

```

**Comment by desperateCoder** [comment]
Hi, thanks for your report!

I lately had something similar,does restarting the app fix this? Worked out for me... If that doesn't help, please try deleting the app storage via android app settings, then retry.

Sorry for the inconveniences though!

While this is a quick fix, I'll investigate this further, seems like an inconsistent answer from the server, really weird...

**Comment by stefan-niedermann** [comment]
@hjuergens will be fixed in `1.14.6`.
